R/mappings.R

```diff
@@ -72,7 +72,18 @@ get_mappings <- function(p, local_only = FALSE) {
 
 #' @export
 get_mappings.ggplot <- function(p, local_only = FALSE) {
-  p$mapping
+  aes_map <- p$mapping
+
+  if (local_only) {
+    return(aes_map)
+  }
+
+  layer_maps <- purrr::map(p$layers, "mapping")
+  intersection <- intersect_with_attr(layer_maps)
+
+  aes_map <- c(aes_map, intersection)
+  class(aes_map) <- "uneval"
+  aes_map
```

Comment:
This smells like something that needs more explanation. I can tell from reading this that you learned something about the structure of a ggplot object that required special handling here — mappings are stored differently in the layer than they are in the base plot? A comment here would be helpful and even better if paired with a test that covers the need for these lines. That way, new readers (like me!) can benefit from what you've learned.